From 80d67294fdbb22154d04ddf2f814c810c9d36f50 Mon Sep 17 00:00:00 2001
From: Andy Ross <andrew.j.ross@intel.com>
Date: Mon, 14 Jul 2014 11:59:04 -0700
Subject: [PATCH 076/117] ov5693: Swap power rail order

Bring 1.8v up/down "inside" of 2.8v to better conform to OV5693
datasheet power sequencing.

Signed-off-by: Andy Ross <andrew.j.ross@intel.com>
---
 drivers/external_drivers/camera/drivers/media/i2c/ov5693.c | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/drivers/external_drivers/camera/drivers/media/i2c/ov5693.c b/drivers/external_drivers/camera/drivers/media/i2c/ov5693.c
index 5e47b02..8166b40 100644
--- a/drivers/external_drivers/camera/drivers/media/i2c/ov5693.c
+++ b/drivers/external_drivers/camera/drivers/media/i2c/ov5693.c
@@ -951,15 +951,15 @@ static int power_ctrl(struct v4l2_subdev *sd, bool flag)
 		return dev->platform_data->power_ctrl(sd, flag);
 
 	if (flag) {
-		ret = dev->platform_data->v1p8_ctrl(sd, 1);
+		ret = dev->platform_data->v2p8_ctrl(sd, 1);
 		if (ret == 0) {
-			ret = dev->platform_data->v2p8_ctrl(sd, 1);
+			ret = dev->platform_data->v1p8_ctrl(sd, 1);
 			if (ret)
-				ret = dev->platform_data->v1p8_ctrl(sd, 0);
+				ret = dev->platform_data->v2p8_ctrl(sd, 0);
 		}
 	} else {
-		ret = dev->platform_data->v2p8_ctrl(sd, 0);
-		ret |= dev->platform_data->v1p8_ctrl(sd, 0);
+		ret = dev->platform_data->v1p8_ctrl(sd, 0);
+		ret |= dev->platform_data->v2p8_ctrl(sd, 0);
 	}
 
 	return ret;
-- 
1.9.3

